### DISCUSSION TITLE
Trace,Logs and Memory Maps in Box64 (with Wine)

### DISCUSSION THREAD

**Discussion by GIOFATT** [ptitSeb/box64#1995] [Q&A]
Is there a way to get the logs, the instruction trace and the memory map of Box64? (I tried changing the env options but it didn't work properly)
Assuming that there is a way to get all of the above, let's pretend Box64 is running Wine and Wine is running a test.exe, is there anything that can help me distinguish which trace belongs to which program? (for example: a specific giveaway instruction or a predefined memory address...)
Is there any additional tool that could help me get the memory map and the instruction trace of Box64 running Wine?

(I am running Wine on Box64 in Amd64 Ubuntu)
Thanks in advance.

**Comment by ptitSeb** [comment]
For the memory map, not yet. I'll probably comes with something soon, but that's a complex question actually.

The "trace" yes, but you probably want a Dump instead. Unfortunatly, this is a dynarec only feature and there is no dynarec for Amd64. A trace need a build of box64 with the "HAVE_TRACE" compile switch enable and then need a old-ish version of libZydis.so for it to work. This might evolve in the future but for now, that's the pre-requesite.
Anyway, a trace to run a wine exe is quite useless, as it needs several thousand of opcode run before anything interesting happens.

To distinguis what is what, you can use `BOX64_TRACE_FILE=$(pwd)/trace-%pid.txt`. The "%pid" will be replace by the actual `pid` of the process and you have have several file, each for it's own process. The default `BOX64_LOG=1` should gives you enough info in the top of the file to understand what is what.

As a workaround for the lack of memory map, you can use `BOX64_JITGDB` (granted the program actually segfault), to trap the segfault in a gdb and then usr `proc/XXX/maps` to have a view on the current porgram being debugging.

Debugging Wine inside Box64 is pretty tricky because box64 has no information on the windows program beeing debugging (it's just code in anonymous memory map), it has symbols and such only for the linux part of wine itself. Again, I plan to improve this situation later, but it's not in the immediate todo....
